# Week 5: ATAC-seq

In these 3 lectures you will learn about [chromatin accessibility](https://en.wikipedia.org/wiki/Epigenomics#Chromatin_accessibility_assays) and the [ATAC-seq protocol](https://en.wikipedia.org/wiki/ATAC-seq). We will go through a few of the first steps for analyzing ATAC-seq data, and a few types of downstream analysis.

## Assignment

The [assignment](assignment.md) consists of questions for 2 tutorials: The [first_steps_tutorial.md](first_steps_tutorial.md), which we will go through in class during Lecture 1, and the [R_analysis_tutorial.md](R_analysis_tutorial.md), which we will go through in class during Lecture 3.

## Lecture 1: ATAC-seq protocol and basic processing

### Topics
- Introduction to chromatin accessibility
- Challenges of studying regulatory DNA
- Introduction to ATAC-seq protocol
- Single-cell ATAC-seq
- Basic steps of initial ATAC-seq analysis

### Resources
- Slides: http://databio.org/slides/open_chromatin.html
- Tutorial: [first_steps_tutorial.md](first_steps_tutorial.md)

## Lecture 2 - ATAC-seq pipeline processing

### Topics
- Pipelines
- Pipeline steps
- The problem of NuMts
- Serial alignments and mtDNA
- Quality control outputs
	- TSS enrichment plot
	- Fragment distribution
	- Partition distribution

### Resources
- Slides: http://databio.org/slides/pepatac.html
- Tutorial: (pending)

## Lecture 3 - ATAC-seq downstream processing

### Topics
- Differential analysis
- Visualization
- Motif analysis
- Footprinting
- Clustering
- Enrichment analysis: GREAT and LOLA

### Resources
- Slides: (pending)
- Tutorial: [R_analysis_tutorial.md](R_analysis_tutorial.md)
